### PR TITLE
Fix flake package

### LINKED ISSUE #27: Broken flake.nix
```
error: builder for '/nix/store/1cxlj18q2vjj09p2dyjk862ssvsp3xwr-rust-workspace-deps-unknown.drv' failed with exit code 101;
       last 10 log lines:
       >   failed to clone into: /build/dummy-src/.cargo-home/git/db/jsonrpc-22a65938f7d26b63
       >
       > Caused by:
       >   network failure seems to have happened
       >   if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
       >   https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
       >
       > Caused by:
       >   failed to resolve address for github.com: Name or service not known; class=Net (12)
       > [naersk] cargo returned with exit code 101, exiting
       For full logs, run 'nix log /nix/store/1cxlj18q2vjj09p2dyjk862ssvsp3xwr-rust-workspace-deps-unknown.drv'.
error: 1 dependencies of derivation '/nix/store/54mlyvashp47rlqysdr1b6b1vqskmv2p-rust-workspace-unknown.drv' failed to build
```
Consider using this instead: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file

### PR DESCRIPTION
**Pull request by nrdxp**
This isn't ideal for development, since it pulls Helix from GitHub instead of from the underlying filesystem, but it does get the package working for nix users. As an additional convenience, I also embedded the runtime by default.

Opening as a draft as I'm not sure if this is the direction we want to take here, but unless NixOS/nix#3978 get's a proper solution, this may be the best way to do it for now.

Potentially resolves #27

### CODE REVIEW

**Comment by pickfire** [comment]
Related pull request https://github.com/helix-editor/helix/pull/180

File: flake.nix [hunk 1]
```diff
@@ -6,6 +6,12 @@
     flake-utils.url = "github:numtide/flake-utils";
     rust-overlay.url = "github:oxalica/rust-overlay";
     naersk.url = "github:nmattia/naersk";
+    helix = {
+      flake = false;
+      url = "https://github.com/helix-editor/helix";
+      type = "git";
```
Comment (gytis-ivaskevicius) [inline flake.nix:12]: Consider replacing it with `url = github:helix-editor/helix;`

File: flake.nix [hunk 2]
```diff
@@ -25,7 +31,8 @@
       in rec {
         packages.helix = naerskLib.buildPackage {
           pname = "helix";
-          root = ./.;
+          root = inputs.helix;
```
Comment (gytis-ivaskevicius) [inline flake.nix:34]: There really must be some better way? :/

**Review by gytis-ivaskevicius — APPROVED** [review]

File: flake.nix (hunk 2 above)
Comment (archseer) [reply flake.nix:34]: Needs fixing in Nix :/ https://github.com/NixOS/nix/issues/4423

**Comment by archseer** [comment]
I think this looks good, it's not ideal but at least it builds :)

File: flake.nix (hunk 1 above)
Comment (nrdxp) [reply flake.nix:12]: Thanks for the review. I actually tried that, as well as `"git:https://..."`, but for some reason it on doesn't understand the submodules attribute when doing so. This is the only way I got it to work.

File: flake.nix (hunk 2 above)
Comment (nrdxp) [reply flake.nix:34]: I did try a `symlinkJoin` and it mostly worked actually. But not if you add new submodules, and it seemed kinda hacky so I left it out.

Comment (gytis-ivaskevicius) [reply flake.nix:34]: Wouldn't symlinkJoin allow avoiding `inputs` altogether and would build package from cloned version?

Comment (nrdxp) [reply flake.nix:34]: How do you mean?

Comment (gytis-ivaskevicius) [reply flake.nix:34]: Ahhh, I see what you did there :D I'd say that it should be left as it is

**Comment by nrdxp** [comment]
I suppose the only question left then, is if the url should point at master as I currently have it, or to the latest release tag?

**Comment by nrdxp** [comment]
I removed the embedded runtime call, since it didn't seem to function as I expected, and I also added an overlay, so users can easily install into a NixOS configuration as well.

**Comment by archseer** [comment]
Sidenote: There's also a proposal to nixpkgs (https://github.com/NixOS/nixpkgs/pull/126208) by @yusdacra, I wonder why the tests fail there

**Review by archseer — APPROVED** [review]

**Pull request merged**